src/ServiceControl.Transports.AzureStorageQueues/QueueLengthProvider.cs

```diff
@@ -38,19 +37,23 @@ public void Process(EndpointInstanceId endpointInstanceId, EndpointMetadataRepor
             var queueName = QueueNameSanitizer.Sanitize(metadataReport.LocalAddress);
 
             var queueClient = CloudStorageAccount.Parse(connectionString).CreateCloudQueueClient();
-            var queue = queueClient.GetQueueReference(queueName);
 
-            queues.AddOrUpdate(endpointInputQueue, _ => queue, (_, currentQueue) =>
+            var queueLength = new QueueLengthValue
             {
-                if (currentQueue.Name.Equals(queue.Name) == false)
+                QueueName = queueName,
+                Length = 0,
+                QueueReference = queueClient.GetQueueReference(queueName)
+            };
+
+            queueLengths.AddOrUpdate(endpointInputQueue, _ => queueLength, (_, currentLength) =>
```

Comment:
I find this line confusing because of multiple usage of `_`. If it means "ignore" it's ok, but just confusing visually.